net/connection.h

```diff
@@ -114,14 +133,17 @@ class Connection : public Threadable
     {
         Command command;
         do {
-            command = m_Socket.readCommand();
+            command = readCommand();
         } while (isRunning() && parseCommand(command));
     }
 
 private:
     std::map<std::string, CommandHandler> m_CommandHandlers;
-    std::unique_ptr<std::mutex> m_SendMutex;
+    std::vector<char> m_Buffer;
     Socket m_Socket;
+    std::unique_ptr<std::mutex> m_SendMutex;
```

Comment:
How come this is a pointer?

Reply:
Sorry, merged without giving you a chance to look at it!

I made the mutex into a pointer so that the class is moveable, which we need so that we can return a ``Connection`` object from a function in the ``Server`` class (although the move might get elided anyway). It turns out mutexes aren't copyable or moveable.